OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs

```diff
@@ -143,29 +148,60 @@ public Actor[] Reinforce(Player owner, string[] actorTypes, CPos[] entryPath, in
 			}
 			else
 			{
-				var aircraftInfo = transport.TraitOrDefault<Aircraft>();
-				if (aircraftInfo != null)
+				var aircraft = transport.TraitOrDefault<Aircraft>();
+				if (aircraft != null)
 				{
-					if (!aircraftInfo.IsPlane)
+					if (!aircraft.IsPlane)
 					{
-						transport.QueueActivity(new Turn(transport, aircraftInfo.Info.InitialFacing));
+						Move(transport, dropoffPoint);
+						transport.QueueActivity(new Turn(transport, aircraft.Info.InitialFacing));
 						transport.QueueActivity(new HeliLand(transport, true));
 					}
 					else
 					{
-						transport.QueueActivity(new Land(transport, Target.FromCell(transport.World, entryPath.Last())));
+						var map = transport.World.Map;
+						var altitude = aircraft.Info.CruiseAltitude.Length;
+						var dropoffCenter = map.CenterOfCell(dropoffPoint);
+						var speed = aircraft.Info.Speed;
+
+						// Distance from the altitude to the ground
```

Comment:
🚨 DUPLICATION ALERT 🚨 

Please adapt `ReturnToBase` to cover your use case, don't just copy all of its code here!